src/Compilers/Test/Utilities/CSharp/TestOptions.cs

```diff
@@ -73,9 +72,16 @@ public static CSharpParseOptions WithTuplesFeature(this CSharpParseOptions optio
             return options;
         }
 
-        public static CSharpParseOptions WithNullCheckingFeature(this CSharpParseOptions options)
+        public static CSharpParseOptions WithNullCheckingFeature(this CSharpParseOptions options, bool includeNonNullChecking = true)
         {
-            return options.WithFeatures(options.Features.Concat(new[] { new KeyValuePair<string, string>("staticNullChecking", "true") }));
+            var features = new List<KeyValuePair<string, string>>();
+            features.AddRange(options.Features);
+            features.Add(new KeyValuePair<string, string>("staticNullChecking", "true"));
+            if (includeNonNullChecking)
+            {
+                features.Add(new KeyValuePair<string, string>("staticNonNullChecking", "true"));
+            }
+            return options.WithFeatures(features);
```

Comment:
It looks like one feature is dependent on the other. What happens if I set one, but not the one it depends on?